extensions/wikia/Blogs/BlogLockdown.php

```diff
@@ -152,4 +152,23 @@ public static function userCan( $title, $user, $action, &$result ) {
 
 		return $return;
 	}
+
+	/**
+	 * Checks permission before delete action on blog articles
+	 *
+	 * @param Article $article
+	 * @param Title $title
+	 * @param User $user
+	 * @param array $permission_errors
+	 *
+	 * @return boolean because it's a hook
+	 */
+	public static function onBeforeDeletePermissionErrors( &$article, &$title, &$user, &$permission_errors ) {
+		// Only users with delete permission can delete a blog article
+		if ( $title->getNamespace() == NS_BLOG_ARTICLE && !$user->isAllowed( 'delete' ) ) {
```

Comment:
The "delete" permission works like the "super delete" permission in MW, so if there is no fine-grained permission to delete a specific type of content, then it falls back on "delete"; It is given to a few groups which you find [here](https://github.com/Wikia/config/blob/dev/PermissionsDefinesBeforeWikiFactory.php) if you search for `'delete'`.